scripts/schemagen.go

```diff
@@ -0,0 +1,281 @@
+// Generates an initial version of a schema for a new resource type.
+//
+// This script draws heavily from https://github.com/radeksimko/terraform-gen,
+// but uses GCP's discovery API instead of the struct definition to generate
+// the schemas.
+//
+// This is not meant to be a definitive source of truth for resource schemas,
+// just a starting point. It has some notable deficiencies, such as:
+// 	* No way to differentiate between fields that are/are not updateable.
+// 	* Required/Optional/Computed are set based on keywords in the description.
+//
+// Usage example (from root dir):
+//
+//   go run ./scripts/schemagen.go -api pubsub -resource Subscription -version v1
+//
+// This will output a file in the directory from which the script is run named `gen_resource_[api]_[resource].go`.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/discovery/v1"
+)
+
+func main() {
+	api := flag.String("api", "", "api to query")
+	resource := flag.String("resource", "", "resource to generate")
+	version := flag.String("version", "v1", "api version to query")
+	flag.Parse()
+
+	if *api == "" || *resource == "" {
+		flag.PrintDefaults()
+		log.Fatal("usage: go run schemagen.go -api $API -resource $RESOURCE -version $VERSION")
+	}
+
+	// Discovery API doesn't need authentication
+	client, err := google.DefaultClient(oauth2.NoContext, []string{}...)
+	if err != nil {
+		log.Fatal(fmt.Errorf("Error creating client: %v", err))
+	}
+
+	discoveryService, err := discovery.New(client)
+	if err != nil {
+		log.Fatal(fmt.Errorf("Error creating service: %v", err))
+	}
+
+	resp, err := discoveryService.Apis.GetRest(*api, *version).Fields("schemas").Do()
+	if err != nil {
+		log.Fatal(fmt.Errorf("Error reading API: %v", err))
+	}
+
+	fileName := fmt.Sprintf("gen_resource_%s_%s.go", *api, underscore(*resource))
+	f, err := os.Create(fileName)
+	defer f.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fields := generateFields(resp.Schemas, *resource)
+
+	buf := &bytes.Buffer{}
+	err = googleTemplate.Execute(buf, struct {
+		TypeName string
+		Fields   map[string]string
```

Comment:
We don't really have a convention around this, but it would be great to have one 👍  I like the idea. Conventions I've seen applied to existing resources include either what you described (Required -> Optional -> Computed) or alphabetical order.